pyramda/prop.py

```diff
@@ -0,0 +1,8 @@
+from pyramda.function.curry import curry
+
+
+@curry
+def prop(x, obj):
+    if isinstance(obj, dict):
```

Comment:
Certainly!

In Javascript, objects are frequently used both to represent traditional OOP-style objects with methods and encapsulated data and to represent traditional dictionaries with plain data in the form of key-value pairs. There is zero syntactic distinction between these two uses, dot-notation (`obj.key`) and bracket-notation (`obj['key']`) are equivalent.

This is absolutely not the case in Python, where dot-notation ("attribute access") is used for "objects" and bracket-notation ("item access") is used for "dictionaries". This is more of a loose rule, as it's possible to wrap both object-style objects and dictionary-style objects in proxies that flip the meaning of item access and attribute access. But this doesn't change the fact that, fundamentally, `obj.key` and `obj['key']` mean very, very different things in Python and the vast majority of Python objects will respond with completely different semantics.

As a result, I think that Pyramda should make a conscious effort not to mix up these two concepts. If a user wants to access both attributes and items with the same function, they can write a proxy class that overrides either `__getitem__` or `__getattr__` depending on how they want to access values.